pkgs/build-support/substitute/substitute-attrs.nix

```diff
@@ -0,0 +1,40 @@
+{ lib, substitute }:
+
+/*
+  This is a wrapper around `substitute` in the stdenv, designed to be as easy to use as
+  `substituteAll` but have fewer pitfalls.
+
+  Attribute arguments:
+
+  - `srcOrAttrs`: either the path to the file to substitute, or an attrset that fits into `substitute`
+
+  - `attrs`: each entry in this set corresponds to a `--subst-var-by` entry in `substitute`.
+    See https://nixos.org/manual/nixpkgs/stable/#fun-substitute
+
+  Example:
+
+  ```nix
+  { substituteAttrs }:
+
+  substituteAttrs ./greeting.txt { world = "hello"; }
+  ```
+
+  See ../../test/substitute for tests
+*/
+srcOrAttrs: attrs:
```

Comment:
why two arguments rather than just an attribute set?

```nix
substituteAttrs { src = ./the/source; subAttrs = { a = xxx; b = zzz; }; };
```

Reply:
I'd be in favor of using an attribute set instead of curried arguments, which also allows clearing up the src/attrs thing with separately named arguments.